### ISSUE TITLE
AltTab opens slowly after recent update (v6.23.0)

### ISSUE THREAD

**Issue by dennis-damico** [lwouis/alt-tab-macos#1096]
**Describe the bug**

When triggering alttab, it takes about 2 seconds to open. If I re trigger immediately after, it opens at the normal speed again.

**Steps to reproduce the bug**

1. Wait about 30 seconds.
2. Trigger alttab with hotkey (notice it takes a few seconds to open).
3. Close alttab
4. Immediately trigger again (notice this time it opens immediately, as it used to before update).

On macos running big sur. Seems like the update to `6.23.0` was what caused the issue.

**Comment by mrc247-DE** [comment]
I can unfortunately confirm it, with today's update (6.23.0) the old "delay problem" has returned.

macOS 11.5.2

**Comment by slaughtr** [comment]
Can also confirm, though for a temporary workaround setting the Apparition Delay to 1ms "fixes" the issue. Still some weird screen overlay bug when the switcher disappears sometimes, though.

**Comment by mrc247-DE** [comment]
I also can confirm the weird screen overlay bug when the switcher disappears sometimes.

**Comment by dennis-damico** [comment]
FWIW, i'm still seeing the issue when setting the delay to `1ms`

**Comment by xieao100** [comment]
是的 我也遇到了这个问题，首次打开特别慢，只有深度用户才能体验出来，你是深度用户，我也是哈哈，我降级旧版本了

**Comment by harrisongoeldner** [comment]
I am also seeing this issue.

**Comment by kbyun03** [comment]
confirming the issue. The switcher disappears after being idle for a while

**Comment by WorkInProgress-Development** [comment]
Just chiming in to add that this also occurs on Catalina 10.15.7

**Comment by Basster** [comment]
After reading all comments above, I don't think its related to the architecture but I can confirm this on 11.5.2 with an M1 Chip, as well.

**Comment by mariomeissner** [comment]
MacBook Pro 16" 2020, same issue, started happening after update. I played around with the 1ms thing, and it works for a few times but then goes back to being slow.

**Comment by lwouis** [comment]
Can the people in this thread facing the issue [take a look at these gifs here](https://github.com/lwouis/alt-tab-macos/issues/563#issuecomment-690654640), and confirm if the symptoms they experience are similar to one of the 3 scenarios?

What may have happened is that the issue from https://github.com/lwouis/alt-tab-macos/issues/563 started showing more often from v6.23.0. I think it's always been there, but now for some reason it happens more often to more people. But first we should confirm what exactly the symptoms are:

* Is it when the app appear that it appears after a delay?
* Is it after the keys are released that it disappears only after a delay?
* Is it when switching from windows on different Spaces that is takes a while to disappear?

**Comment by WorkInProgress-Development** [comment]
I haven't experienced the second GIF, presumably as I don't use multiple spaces very often these days.
However, the frozen ui effect of the first one has been happening, and the remnants of the previous window from the 3rd gif have both been happening for me.

**Comment by Basster** [comment]
Same here. I don't use spaces, neither, but the third one was noticeable. I didn't related it to AltTab, because I have a new DisplayLink Docking station in use for some days. But since I downgraded to AltTab 6.21.2, I haven't seen it, so I assume it was AltTab related.

**Comment by ipeluffo** [comment]
Agree that I started noticing the behaviour of the 3rd gif since v6.23.0.

That said, IMHO, this issue describes a different issue where alttab takes longer that the delay time set in the preferences. Then even if delay is 0ms or 1ms, alttab is taking around 1 sec to appear after pressing the keys combination (I'm not sure how to make a recording showing the keys that I press).

Also I really appreciate all the hard work on this. Thanks

**Comment by slaughtr** [comment]
3rd gif + mentioned ~1 sec delay in showing the switcher. Though I'm not using multiple spaces, I do use a window manager and leave all items on one screen full screen, so I only see visual issues if a window resizes on my main screen or it switches apps on the other screen. Sorry if that's not super clear.

**Comment by gbabichev** [comment]
Can confirm I'm seeing the 3rd gif. M1 MacBook Air & M1 Mac Mini. Started on 6.23.0

**Comment by Jaxsun** [comment]
I see something similar to the third gif sometimes, however the bigger issue is that when I hit the alt+tab key the switcher just takes over a second to appear. it seems responsive when disappearing however.

**Comment by mariomeissner** [comment]
@lwouis I don't experience any of the three gifs. The answer to your three questions would also be a no. What happens is that when I press Alt+Tab, I have to wait around 1 second until I see the popup appear. I can then smoothly press tab more times to choose the correct app, and it will immediately switch, no issues. The issue for me is only the initial popup appearance delay.

Thank you so much for your hard work, I love this app!

**Comment by GaganN18** [comment]
Yes the 3rd gif is also the case for my m1 mac pro 16gb ram, and also startup time is delayed after not using alt tab for a short while

**Comment by GaganN18** [comment]
using the version just before (6.22.1) works perfectly fine

**Comment by lwouis** [comment]
I profile in Instruments a lot, but like https://github.com/lwouis/alt-tab-macos/issues/563, nothing shows. The delay/freeze is on Apple side, inside their processes, probably `WindowServer`.

In any case, by looking at the diff with the previous version, I think I isolated the code that's an issue. It's very esoteric though, and I couldn't find a real explanation of it online, or tips on how to robustly deal with it. I made a workaround that seems to work on my local machine.

Could you [please try this build here](https://github.com/lwouis/alt-tab-macos/files/7107388/AltTab.app.zip), and share if it fixes it for you? If it does, I'll quickly release an official build to everyone 👍

**Comment by evgkrsk** [comment]
This build resolved problem for me.

**Comment by destronije** [comment]
@lwouis yes, seems like the issue is resolved.

**Comment by gbabichev** [comment]
I tried the new version but am still experiencing lag when displaying the Alt Tab window. It draws partially, then opens up all the way.

**Comment by lwouis** [comment]
@gbabichev is it consistently happening? Could you please share a video? What are the steps you go through to see the issue?

**Comment by lwouis** [comment]
I received a lot of messages from users about the issue with the delay discussed in this ticket. I decided to try to release the build I shared above. Hopefully it fixes the issue. I wish there was more QA done, but it's a tradeoff between waiting for more people to help test, or releasing to the public. I don't want to disappoint with the fix, but I also think better 2 releases than making people wait and deal with the issue, as well as me getting flooded with reports

**Comment by WorkInProgress-Development** [comment]
It's always a fine line between quality assurance and quick fixes, but I feel in this case that was definitely a good choice.

**Comment by lzq1126** [comment]
mac os 1.11.5 still delay main window for v6.24.0 ，ghost window.

**Comment by lwouis** [comment]
Hi @lzq1126,

Please imagine how hard it is to do something about an issue when all i have is 1 line of text.

If you face an issue, please open a new ticket. If it's related this one, you can reference it in the ticket of course. When opening the new ticket, you will be presented with a template that guides you to fill important information for a bug report. Steps to reproduce, video to demonstrate the issue clearly, etc.

I've worked on this project for 2 years now, and i can see clearly that the only thing that matters with bugs is that i have a reliable way to reproduce them on my machine, so i can debug and fix them. Without that step, nothing will get done.

Thank you
